drivers/infiniband/hw/bnxt_re/ib_verbs.c

```diff
@@ -309,7 +309,10 @@ int bnxt_re_del_gid(const struct ib_gid_attr *attr, void **context)
 		 */
 		if (ctx->idx == 0 &&
 		    rdma_link_local_addr((struct in6_addr *)gid_to_del) &&
-		    ctx->refcnt == 1 && rdev->gsi_ctx.gsi_sqp) {
+		    ctx->refcnt == 1 &&
+		    (rdev->gsi_ctx.gsi_sqp ||
+		     rdev->gsi_ctx.gsi_qp->qplib_qp.type ==
+		     CMDQ_CREATE_QP_TYPE_GSI)) {
```

Comment:
There is a diff between how we handle this area in upstream vs OOB.
So i think there is no memleak issue in the upstream code because was are returning EFAULT as opposed to 0 in OOB.